src/main/java/net/neoforged/neoforge/transfer/fluid/FluidUtil.java

```diff
@@ -147,6 +147,27 @@ private static void playSoundAndGameEvent(FluidStack stack, Level level, @Nullab
         // Prioritize block position, use player position as a fallback
         Vec3 position = blockPos != null ? Vec3.atCenterOf(blockPos) : new Vec3(player.getX(), player.getY() + 0.5, player.getZ());
 
+        playSoundAndGameEvent(stack, level, position, player, pickup);
+    }
+
+    /**
+     * Triggers the appropriate sound effect and game event for an interaction with a fluid handler.
+     *
+     * @param resource The resource that was moved during the interaction.
+     * @param level    The level the interaction occurred in.
+     * @param position Where the interaction occurred at. The sound and game event will trigger here.
+     * @param player   The player that caused the interaction (optional). The game event will be attributed to them.
+     * @param pickup   True if the fluid was extracted from the handler, false if it was inserted.
+     */
+    public static void triggerSoundAndGameEvent(FluidResource resource, Level level, Vec3 position, @Nullable Player player, boolean pickup) {
```

Comment:
Mostly duplicated with the following method, no?

Reply:
One is FluidStack based, the other FluidResource.

In a public API, I'd favor FluidResource (going forward, in general). But there's no telling whether a custom FluidType uses the *amount* of fluid moved to influence which sound is played.

What's your thought on that, should I add an `int amount` parameter, or expose the `FluidStack` version?

Reply:
IMO using the amount is not a good idea. So I think using one of the helpers from the other makes sense.